RevenueCat/Plugins/iOS/PurchasesUnityHelper.m

```diff
@@ -265,7 +265,15 @@ - (void)promotionalOfferForProductIdentifier:(NSString *)productIdentifier
     [RCCommonFunctionality promotionalOfferForProductIdentifier:productIdentifier
                                                       discount:discountIdentifier
                                                completionBlock:^(NSDictionary *_Nullable responseDictionary, RCErrorContainer *_Nullable error) {
-        [self sendJSONObject:responseDictionary toMethod:GET_PROMOTIONAL_OFFER];
+        NSMutableDictionary *response;
+        if (error) {
+            response = [NSMutableDictionary new];
+            response[@"error"] = error.info;
+
+        } else {
+            response = [NSMutableDictionary dictionaryWithDictionary:responseDictionary];
```

Comment:
Do we need to make sure that `responseDictionary` isn't `nil` at all? Or is it safe to assume it never will be?

Reply:
Yeah I'd create the `NSMutableDictionary` in line 268 so we don't risk sending `nil`.

Reply:
Initializing to an empty dictionary will break the unity code where we try to create the classes from the json response. I think we should create an error here, but the error should be in the purchases hybrid common side so all hybrids are protected from that unexpected case where both the response and the error are nil

Reply:
Oh actually, for incompatible versions phc sends nil for both 🤔

Reply:
I opened this PR https://github.com/RevenueCat/purchases-hybrid-common/pull/338/ to stop sending two nils, and also created an error in this PR if both responses and error are nil